internal/newusers/manager.go

```diff
@@ -2,17 +2,42 @@
 package newusers
 
 import (
+	"errors"
 	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"os/user"
+	"strconv"
+	"syscall"
 	"time"
+	_ "unsafe" //nolint:revive,nolintlint // needed for go:linkname. nolintlint as false positive then.
 
 	"github.com/ubuntu/authd/internal/newusers/cache"
 	"github.com/ubuntu/authd/internal/users"
 	"github.com/ubuntu/decorate"
 )
 
+const (
+	// defaultEntryExpiration is the amount of time the user is allowed on the cache without authenticating.
+	// It's equivalent to 6 months.
+	defaultEntryExpiration = time.Hour * 24 * 30 * 6
+
+	// defaultCleanupInterval is the interval upon which the cache will be cleaned of expired users.
+	defaultCleanupInterval = time.Hour * 24
+)
+
+// dirtyFlagDbName is the dirty flag exported for tests.
+//
+//go:linkname dirtyFlagDbName github.com/ubuntu/authd/internal/newusers/cache.dirtyFlagDbName
+var dirtyFlagDbName string
+
 // Manager is the manager for any user related operation.
 type Manager struct {
-	cache *cache.Cache
+	cache          *cache.Cache
+	doClear        chan struct{}
+	quit           chan struct{}
+	cleanupQuitted chan struct{}
```

Comment:
`cleanupDone` maybe?

Reply:
I kept the same name that we had before and, IMHO, quitted is better since this channel is to indicate that the routine is no longer running rather than "done".

Reply:
> I kept the same name that we had before and

Yeah, i see... maybe `cleanupFinished` It's just that _quitted_ sounds bad to me :joy:

Reply:
What about `cleanupStopped`?